src/Raven.Client/Documents/Session/Operations/QueryOperation.cs

```diff
@@ -165,34 +165,39 @@ public void EnsureIsAcceptableAndSaveResult(QueryResult result)
             if (result == null)
                 throw new IndexDoesNotExistException("Could not find index " + _indexName);
 
-            if (_indexQuery.WaitForNonStaleResults && result.IsStale)
+            EnsureIsAcceptable(result, _indexQuery.WaitForNonStaleResults, _sp, _session);
+
+            _currentQueryResults = result;
+            //_currentQueryResults.EnsureSnapshot();
+
+            if (Logger.IsInfoEnabled)
             {
-                _sp?.Stop();
-                var msg = $"Waited for {_sp?.ElapsedMilliseconds:#,#;;0} ms for the query to return non stale result.";
+                var isStale = result.IsStale ? "stale " : "";
+                Logger.Info($"Query returned {result.Results.Items.Count()}/{result.TotalResults} {isStale}results");
```

Comment:
Better log message here.

Reply:
fixed in https://github.com/ravendb/ravendb/pull/4681